common/models/role.js

```diff
@@ -289,7 +299,11 @@ module.exports = function(Role) {
     var resolver = Role.resolvers[role];
     if (resolver) {
       debug('Custom resolver found for role %s', role);
-      resolver(role, context, callback);
+
+      var promise = resolver(role, context, callback);
+      if (promise instanceof Promise) {
+        promise.then(callback).catch(callback);
```

Comment:
This version is subtly different from my proposal above: when `callback` throws an error, it will be invoked again with the error thrown:

```js
callback = function(err) {
  console.log(err || 'ok');
  throw new Error('bang!');
}
```

The result: resolver is called only once, but two messages are printed:

```
ok
Error: bang!
```

The implementation I proposed above should treat the error thrown from callback as an unhandled error (promise rejection), which is IMO more correct.